src/server/worker/pipeline/spout/spout.go

```diff
@@ -7,6 +7,13 @@ import (
 
 // Run will run a spout pipeline until the driver is canceled.
 func Run(driver driver.Driver, logger logs.TaggedLogger) error {
+	// If the output branch head commit is open, close it
+	pipelineName := driver.PipelineInfo().Pipeline.Name
+	outputBranch := driver.PipelineInfo().OutputBranch
+	if err := driver.PachClient().FinishCommit(pipelineName, outputBranch, ""); err != nil {
+		return err
+	}
+
```

Comment:
So this is more of an issue now because creating the spout will have a default head commit on the output branch (which is left open because it's a pipeline), but yeah, it's probably good to be defensive here in case the user code leaves a commit open.